src/states/elder/mod.rs

```diff
@@ -1006,49 +943,32 @@ impl Elder {
 
     // If this returns an error, the peer will be dropped.
     fn handle_bootstrap_request(&mut self, pub_id: PublicId) -> Result<(), RoutingError> {
-        let peer_kind = if let Some(peer) = self.peer_mgr.get_peer(&pub_id) {
-            match *peer.state() {
-                PeerState::Bootstrapper { peer_kind, .. } => peer_kind,
-                _ => {
-                    return Err(RoutingError::InvalidStateForOperation);
-                }
+        let conn_info = match self.peer_map().get(&pub_id) {
+            Some(conn_info) => conn_info.clone(),
+            None => {
+                log_or_panic!(
+                    LogLevel::Error,
+                    "Not connected to the sender of BootstrapRequest."
+                );
+                return Err(RoutingError::UnknownConnection(pub_id));
             }
-        } else {
-            return Err(RoutingError::UnknownConnection(pub_id));
         };
 
-        if peer_kind == CrustUser::Client {
-            let ip = self
-                .network_service
-                .get_peer_ip_addr(&pub_id)
-                .map_err(|err| {
-                    debug!(
-                        "{} Can't get IP address of bootstrapper {:?} : {:?}",
-                        self, pub_id, err
-                    );
-                    self.disconnect_peer(&pub_id);
-                    err
-                })?;
-
-            if !self.peer_mgr.can_accept_client(ip) {
-                debug!(
-                    "{} Client {:?} rejected: We cannot accept more clients.",
-                    self, pub_id
-                );
-                self.send_direct_message(
-                    &pub_id,
-                    DirectMessage::BootstrapResponse(Err(BootstrapResponseError::ClientLimit)),
-                );
-                self.disconnect_peer(&pub_id);
-                return Ok(());
-            }
-        }
+        let peer_type = match conn_info {
+            ConnectionInfo::Client { .. } => "client",
+            ConnectionInfo::Node { .. } => "node",
+        };
+        trace!(
+            "{} - Received BootstrapRequest from {:?} as {}.",
+            self,
+            pub_id,
+            peer_type
+        );
 
-        if (peer_kind == CrustUser::Client || !self.is_first_node)
-            && self.chain.len() < self.min_section_size() - 1
-        {
+        // Check min section size.
+        if !self.is_first_node && self.chain.len() < self.min_section_size() - 1 {
```

Comment:
Won't this make the first node accept clients even before the first section grows enough to be fully functional?

Reply:
Yes, but that's what the original (before this PR) behaviour was. 

Maybe the first node should only accept nodes, not clients though (until reaching min section size)? I think before this PR, we weren't able to tell whether the bootstrapping peer is a client or node until the relocation, so we accepted both. But now we know it*, so maybe we can restrict this to nodes only. What do you think?

*) With crust, both client and nodes started as `CrustUser::Cient` and nodes then switched to `CrustUser::Node` after relocation. Now we identify as client or node right from the start which I'm not 100% sure it's correct.

Reply:
>Yes, but that's what the original (before this PR) behaviour was.
Hmm, but before this PR, this `if` would cause even the first node to reject clients if the section was too small, wouldn't it? That would be even weirder with what you are saying, because it would mean that before this PR everyone would be rejected when the section was small. What am I missing? Is it that this happens at some later stage of bootstrapping?

Reply:
Oops, I think I got confused and am spreading misinformation now :grimacing: . Ignore the previous comment from me. You were right. I'm going to modify the code.